pkg/monitor/cluster/hiveregistrationstatus_test.go

```diff
@@ -0,0 +1,270 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hiveclient "github.com/openshift/hive/pkg/client/clientset/versioned"
+	hivefake "github.com/openshift/hive/pkg/client/clientset/versioned/fake"
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/hive"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+)
+
+func TestEmitHiveRegistrationStatus(t *testing.T) {
+	fakeNamespace := "fake-namespace"
+
+	for _, tt := range []struct {
+		name       string
+		oc         *api.OpenShiftCluster
+		cd         kruntime.Object
+		withClient bool
+		wantErr    string
+		wantLog    string
+	}{
+		{
+			name:       "no hiveclient",
+			withClient: false,
+			wantLog:    "skipping: no hive cluster manager",
+		},
+		{
+			name:       "No Namespace",
```

Comment:
Nit: just for consistancy in caps with other test case names.
```suggestion
			name:       "no namespace",
```

Reply:
is lower case now